src/emulator/storage/persistence.ts

```diff
@@ -67,17 +81,25 @@ export class Persistence {
         if (err) {
           reject(err);
         } else {
+          this._diskPathMap = new Map();
           resolve();
         }
       });
     });
   }
 
   renameFile(oldName: string, newName: string): void {
-    renameSync(this.getDiskPath(oldName), this.getDiskPath(newName));
+    const oldNameId = this._diskPathMap.get(oldName)!;
```

Comment:
This is actually a little bit more nuanced than I thought it was. The way our upload scheme works, is that in-progress uploads get stored to [tmp staging directory](https://github.com/firebase/firebase-tools/blob/master/src/emulator/storage/upload.ts#L113), so we need to actually move the file to the proper emulator directory here.

That being said, because in progress uploads are not being tracked by `Files._files`, we don't technically need a staging directory since only tracked files are exported. 

In short, could you remove the staging directory logic in another PR (with tests) so that we can maintain the logic of changing the virtual pointer here (instead of actually moving the file on disk)?

Reply:
From what I could tell `Upload.path` actually just a filename is in the format of `${uploadId}_b_${bucketId}_o_${objectId}` and isn't actually pointing to a different directory. 

In all cases where `Upload.path` is used it is going through the Persistence layer and bing mapped to a virtual pointer too. So generally even during the staging step, files should be stored in the final emulator directory and there shouldn't be a staging directory. 

I think the map looks like 
`{uploadId_b_bucketId_o_objectId: dirPath/UUID}`
during staging and should look like `{bucketId/objectId: dirPath/UUID}` after, which this code should be able to handle. Ultimately I'm not sure if the suggested changes are needed since there is no temp staging directory.

Please let me know if I'm missing something and there are changes that need to be made here

Reply:
synced offline, you're right I was misremembering the current state of the code and spoke before I verified my statements. There's no separate staging dir so this pattern still works as is